sycl/include/sycl/ext/oneapi/experimental/sycl_complex.hpp

```diff
@@ -1010,6 +1010,464 @@ __DPCPP_SYCL_EXTERNAL _SYCL_EXT_CPLX_INLINE_VISIBILITY
 } // namespace oneapi
 } // namespace ext
 
+template <typename T> using marray_data = sycl::detail::vec_helper<T>;
+
+template <typename T>
+using marray_data_t = typename detail::vec_helper<T>::RetType;
+
+template <typename T, std::size_t NumElements>
+class marray<sycl::ext::oneapi::experimental::complex<T>, NumElements> {
```

Comment:
That sounds great! I actually mentioned and proposed that in a draft PR https://github.com/intel/llvm/pull/10854

Would it be better for the `complex` and `marray`, to live in a directory `complex` under `sycl/include/sycl/ext/oneapi/experimental/` where one file will be the `complex` header, one file will be the `marray` header and one would be the common details: helper functions, traits ... that are shared by the complex headers?

Otherwise for simplicity, I'll just split the two and include complex into marray.

What do you think?

Reply:
I am okay with both! The former may mean less inter-dependencies which may make it easier to make changes to in the future, so slight preference for that from my side. 😄